### PR TITLE
Put request handler on SimpleServer.prototype

### PR DESCRIPTION
**Pull request by Arlen22**
The request handler may be used by ExpressJS apps directly and can do most of the heavy lifting without any modification. Note that the self variable must be assignee using `[Function].bind(null,SimpleServer instance)`.

This should be tested first to make sure the --server command works as expected. If so, then it should be good. I dont know what tests you normally run.

### CODE REVIEW

**Comment by Arlen22** [comment]
Crossposted from #2607:

Ok, I tested this on my Android phone using Termux, and it works there. The code itself never uses `this`, only `self`, and this is how I did it before, so I think it will work.

File: core/modules/commands/server.js [hunk 1]
```diff
@@ -141,7 +139,11 @@ SimpleServer.prototype.listen = function(port,host) {
 				});
 				break;
 		}
-	}).listen(port,host);
+	};
+	
+SimpleServer.prototype.listen = function(port,host) {
+	var self = this;
+	http.createServer(self.requestHandler.bind(null,self)).listen(port,host);
```
Comment (sukima) [inline core/modules/commands/server.js]: There is no inner scope so `this` works without `self`. Also if `requestHandler` never uses `this` then there is no need to `bind` Also it removes the need to be on the `SimpleServer.prototype`.

What if requestHandler dropped the `self` and used `this`?

```js
http.createServer(this.requestHandler.bind(this)).listen(port,host);
```

File: core/modules/commands/server.js [hunk 2]
```diff
@@ -91,57 +91,59 @@ SimpleServer.prototype.checkCredentials = function(request,incomingUsername,inco
...
-		}
-	}).listen(port,host);
+			});
+			break;
+	}
+};
+	
+SimpleServer.prototype.listen = function(port,host) {
+	var self = this;
+	http.createServer(self.requestHandler.bind(null,self)).listen(port,host);
```
Comment (sukima) [inline core/modules/commands/server.js]: There is no inner scope so `this` works without `self`. Also if `requestHandler` never uses `this` then there is no need to `bind` Also it removes the need to be on the `SimpleServer.prototype`.

What if requestHandler dropped the `self` and used `this`?

```js
http.createServer(this.requestHandler.bind(this)).listen(port,host);
```

Comment (Arlen22) [inline core/modules/commands/server.js]: Yes, you are right about `this`, however it still needs to be the first variable.

**Comment by Arlen22** [comment]
The reason I left it there is just so as to not change anything in any way beyond the minimum required.

**Comment by Arlen22** [comment]
The diff is a pain, but basically all I did is pull the request handler out into a separate property of the prototype so it could be called directly. This allows ExpressJS to use it as a regular (but final) route. Vanilla HTTP server would just be a matter of calling the handler of the desired instance with the request and response arguments.

**Comment by Arlen22** [comment]
In thinking about it, I decided to do it to simplify matters. `self` is used everywhere, so if anything is depending on `this`, that is probably a bug. 

@Jermolene, is there anything further that I can do to help with this, or is this just a matter of waiting for the next release cycle. Let me know if I'm missing anything.

**Comment by Jermolene** [comment]
Hi @Arlen22 my apologies, I think this is good to go.

**Comment by linonetwo** [comment]
So is there an example to set up express to server tiddlywiki?

I found `root-tiddler=$:/core/save/lazy-images` very slow, using express to do HTTP2 might help.

**Comment by linonetwo** [comment]
Ouch, it seems things in this PR no longer exist...

**Comment by Jermolene** [comment]
Hi @linonetwo this was merged some time ago, and there's been a fair amount of refactoring since then.  @Arlen22 may be able to help on your question of using express to serve TiddlyWiki.

**Comment by Arlen22** [comment]
@linonetwo The best example would probably be in the guts of TiddlyServer, as that is exactly what TiddlyServer does (without Express). You can find the relevant code in https://github.com/Arlen22/TiddlyServer/blob/v2.2/src/data-folder.ts. I'm currently refactoring this branch so some things may change, but this one is mostly done.

**Pull request merged**